state/azure/cosmosdb/cosmosdb.go

```diff
@@ -208,3 +265,57 @@ func (c *StateStore) BulkDelete(req []state.DeleteRequest) error {
 
 	return nil
 }
+
+// Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail
+func (c *StateStore) Multi(operations []state.TransactionalRequest) error {
+	upserts := []CosmosItem{}
+	deletes := []CosmosItem{}
+
+	partitionKey := "UNKNOWN"
+	for _, o := range operations {
+		if o.Operation == state.Upsert {
+			req := o.Request.(state.SetRequest)
+
+			partitionKey = populatePartitionMetadata(req.Key, req.Metadata)
```

Comment:
does it needs a check that partition key for all requests should be same and then return err if anyone of them is different because cosmos will not guarantee transaction  if partitionkeys are different

Reply:
Cosmos will fail the stored procedure if the partition keys differ, but I added an earlier check here.